SECURITY.md

```diff
@@ -29,3 +35,89 @@ If you are editing this document please feel encouraged to change this format to
 - #993 Add restricted permissions to RocksDB - 1.4.5 - Anonymous security review
 - #740 Add hardening compile flags and `-fPIE` - 1.4.1 - Anonymous security review
 - #300 Add restricted permissions to osqueryd logging - 1.0.4
+
+## Security Requirements Met by Functionality
+
+### Confidentiality
+
+#### No Retrieval of Sensitive User-owned Data
+
+- Certain tables disabled by default, additional tables can be disabled
+
+#### Use of HTTPS (TLS)
+
+- HTTPS (specifically the TLS v1.2 protocol) is used to encrypt all communications
+
+#### No custom cryptography (only well-known libraries)
+
+#### No network connectivity to third parties
+
+#### Does not accept incoming network connections
+
+#### Does not 'shell out' to other binaries for data collection
+
+#### Access control functions
+
+### Integrity
+
+#### Limited attack surface
+
+#### Daemon runs as root
+
+- Its files are owned as root (config, RocksDB) to resist tampering
+- Extensions must also run as root, communicate over IPC restricted to root
+
+#### No self-update feature
+
+#### No methods to modify data on the system
+
```

Comment:
This is something osquery seeks to achieve but it is best effort. We strongly suspect this to be true and if we are made aware of invariant we treat them as high priority bugs and promptly fix them.